services/dedup/badger/badger.go

```diff
@@ -96,10 +95,11 @@ func NewBadgerDB(conf *config.Config, stat stats.Stats, path string) *Dedup {
 	db.stats.vlogSize = stat.NewTaggedStat("badger_db_size", stats.GaugeType, stats.Tags{"name": "dedup", "type": "vlog"})
 	db.stats.totSize = stat.NewTaggedStat("badger_db_size", stats.GaugeType, stats.Tags{"name": "dedup", "type": "total"})
 
-	return &Dedup{
-		badgerDB:    db,
-		uncommitted: make(map[string]struct{}),
+	err := db.init()
```

Comment:
since we are now calling `init` in the producer function, we no longer need the `sync.Once`